test/integration/user-management-api/applications_test.rb

```diff
@@ -34,255 +34,280 @@ def setup
+      get admin_api_applications_path, params: params.merge({ service_id: @service.id })
 
-    def access_token_params(token = @token)
-      { access_token: token.value }
+      assert_response :success
+      assert_applications_count @response.body, @service.cinstances.size
     end
 
-    alias params access_token_params
+    test '#index with access to no service' do
+      User.any_instance.expects(:member_permission_service_ids).returns([]).at_least_once
+      get admin_api_applications_path, params: params.merge({ service_id: @service.id })
+
+      assert_response :success
+      assert_applications_count @response.body, 0
+    end
   end
 
   class ProviderKeyTest < EnterpriseApiApplicationsTest
-    context 'total_entries <= per_page' do
-      should 'pagination is off' do
-        get admin_api_applications_path(format: :xml), params: params
+    protected
 
-        assert_response :success
-        assert_not_pagination @response.body, "applications"
-      end
+    def private_key_params
+      { provider_key: @provider.api_key }
     end
 
-    context 'total_entries > per_page' do
-      setup do
-        application_plan = FactoryBot.create(:application_plan, issuer: @provider.first_service!)
-        @buyer.buy! application_plan
-        application_plan2 = FactoryBot.create(:application_plan, issuer: @provider.first_service!)
-        @buyer.buy! application_plan2
-        @max_per_page = set_api_pagination_max_per_page(to: 1)
+    alias params private_key_params
+
+    def target
+      { id: @application.id, user_account_id: @buyer.id, application_id: @application.application_id }
+    end
+  end
+
+  class PaginationTest < ProviderKeyTest
+    def setup
+      super
+      @service = FactoryBot.create(:service, account: @provider)
+      @total_entries = 5
+      @total_entries.times do
+        @buyer.buy! FactoryBot.create(:application_plan, issuer: @service)
       end
+      @max_per_page = set_api_pagination_max_per_page(to: 4)
+    end
 
-      attr_reader :max_per_page
+    def teardown
+      reset_pagination_config!
+    end
 
-      should 'index is paginated' do
-        get admin_api_applications_path(format: :xml), params: params.merge({ per_page: 1 })
+    attr_reader :total_entries, :max_per_page
 
-        assert_response :success
-        assert_pagination @response.body, "applications"
-      end
+    test 'pagination is off if total_entries <= per_page' do
+      set_api_pagination_max_per_page(to: total_entries)
 
-      should 'pagination per_page has a maximum allowed' do
-        get admin_api_applications_path(format: :xml), params: params.merge({ per_page: max_per_page + 1 })
+      get admin_api_applications_path(format: :xml), params: params
+      assert_response :success
 
-        assert_response :success
-        assert_pagination @response.body, "applications", per_page: max_per_page
-      end
+      assert_applications_count @response.body, total_entries
```

Comment:
Actually where is this defined? Tests fail with no such method.

Reply:
Yeah I put it into a stash and forgot about it 😅